src/mococrw/util.h

```diff
@@ -45,6 +45,15 @@ std::vector<uint8_t> fromHex(const std::string &hexData);
 
 std::vector<uint8_t> cryptoRandomBytes(size_t length);
 
+/**
+ * @brief Transforms given string to a string with percent encoding notation (see RFC 3986)
+ * i.e. "30AFF" -> "%03%0A%FF"
+ * @note This function does not enforce a string to contain only hex characters (only hex digits
+ * are used in percent encoding). If you have a different use case, change this or create your
+ * own function.
+ */
+void hexStringToPctEncoded(std::string &str);
```

Comment:
I don't think that this is a generally useful function. In URIs only non-ascii characters are escaped via this ampersand notation. The function here does no particular encoding but just adds an % every second character.  We should have this as a private function. Moreover, functions with side-effects are generally discouraged. This should be a string -> string function.